src/space.cpp

```diff
@@ -843,8 +843,11 @@ static argval_t do_space(chunk_t *first, chunk_t *second, int &min_sp, bool comp
          log_rule("sp_template_angle");
          return(cpd.settings[UO_sp_template_angle].a);
       }
-      log_rule("sp_before_angle");
-      return(cpd.settings[UO_sp_before_angle].a);
+      if (first->type != CT_QUALIFIER)
+      {
+         log_rule("sp_before_angle");
+         return(cpd.settings[UO_sp_before_angle].a);
```

Comment:
Why is this if check necessary?

Reply:
Is the change accompanied by a test?

Reply:
Yes, 018e14b9e24e9f5d8bb88b0d17ec83fa13db299d